server/routes/userRoutes.js

```diff
@@ -106,18 +93,56 @@ module.exports = (db) => {
     const { host_name } = req.body.params;
     dbHelper
       .getWebsiteIDByHostname(host_name)
-      // .getWebsiteIDByHostname("reddit.com")
-      .then((site) => dbHelper.addWebsiteToBlacklist(userId, site.id))
+      .then((site) => {
+        if (!site) {
+          // Removes '.com' from end of hostname for DB.
+          const remSuffix = host_name.split(".")[0];
+          // Capitalizes first letter of hostname
+          const name = remSuffix.charAt(0).toUpperCase() + remSuffix.slice(1);
+          const category = null;
+          dbHelper
+          .addWebsite(host_name, name, category)
+          .then((site) => {
+            dbHelper.addWebsiteToBlacklist(userId, site.id);
+          });
+        }
+        dbHelper.addWebsiteToBlacklist(userId, site.id);
```

Comment:
This needs to be put into an `else` block (as part of the `if (!site)`)

Reply:
I had it previously to if(!site.id), which made line 109 implicit that site.id is truthy

Reply:
Your function to remove the suffix actually takes the www and capitalizes it - you want index 1 to get the actual host name.

Reply:
Actually, I think that would depend on how it was sent from the client - like how we want to handle websites that are typed www.google.com, or google.com, or http:// etc

Reply:
He has it correct... we are storing urls as google.com, no subdomains